alpha/action/migrations/migrations.go

```diff
@@ -0,0 +1,122 @@
+package migrations
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+type MigrationToken string
+
+const MigrationTokenInvalid MigrationToken = ""
+
+type Migration interface {
+	Token() MigrationToken
+	Name() string
+	Help() string
+	Migrate(*declcfg.DeclarativeConfig) error
+}
+
+func newMigration(name string, help string, fn func(config *declcfg.DeclarativeConfig) error) Migration {
+	s := strings.ReplaceAll(help, "`", "\"")
+	return &simpleMigration{token: MigrationToken(name), name: name, help: s, fn: fn}
+}
+
+type simpleMigration struct {
+	token MigrationToken
+	name  string
+	help  string
+	fn    func(*declcfg.DeclarativeConfig) error
+}
+
+func (s simpleMigration) Token() MigrationToken {
+	return s.token
+}
+
+func (s simpleMigration) Name() string {
+	return s.name
+}
+
+func (s simpleMigration) Migrate(config *declcfg.DeclarativeConfig) error {
+	return s.fn(config)
+}
+
+func (s simpleMigration) Help() string {
+	return s.help
+}
+
+type Migrations struct {
+	Migrations []Migration
+}
+
+func GetLastMigration() MigrationToken {
+	if len(allMigrations) == 0 {
+		return ""
+	}
+	return allMigrations[len(allMigrations)-1].Token()
+}
+
+func ValidateName(name string) (MigrationToken, error) {
+	for _, migration := range allMigrations {
+		if migration.Name() == name {
+			return migration.Token(), nil
+		}
+	}
+	return MigrationTokenInvalid, fmt.Errorf("unknown migration level %q", name)
+}
+
+// allMigrations represents the migration catalog
+// the order of these migrations is important
+var allMigrations = []Migration{
+	bundleObjectToCSVMetadata,
+}
+
+func NewMigrations(level MigrationToken) (*Migrations, error) {
+	if level == "" {
+		return &Migrations{}, nil
+	}
+
+	migrations := slices.Clone(allMigrations)
+
+	found := false
+	keep := migrations[:0]
+	for _, migration := range migrations {
+		keep = append(keep, migration)
+		if migration.Token() == level {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return nil, fmt.Errorf("unknown migration level %q", level)
+	}
+	return &Migrations{Migrations: keep}, nil
+}
+
+func HelpText() string {
+	var help strings.Builder
+	help.WriteString("\nThe migrator will run all migrations up to and including the selected level.\n\n")
+	help.WriteString("Available migrators:\n")
+	if len(allMigrations) == 0 {
+		help.WriteString("   (no migrations available in this version)\n")
+	}
+
+	tabber := tabwriter.NewWriter(&help, 0, 0, 1, ' ', 0)
+	for _, migration := range allMigrations {
+		fmt.Fprintf(tabber, "  - %s\t: %s\n", migration.Name(), migration.Help())
```

Comment:
We can use `migration.Token()` without other changes:
```suggestion
		fmt.Fprintf(tabber, "  - %s\t: %s\n", migration.Token(), migration.Help())
```

Go is able to figure out that `MigrationToken` is a string, so when it interprets it with `%s` formatting, it just uses the raw underlying string. If, in the future we need a special "token" to "name" function for some reason, I think we can define `String()` of the `MigrationToken` type and put whatever translation logic we want.

Reply:
If we do this, I think we can remove the `name` field and `Name()` method related to a migration.